rpc/src/eth_rpc.rs

```diff
@@ -507,15 +507,12 @@ impl EthApiServer for KakarotEthRpc {
         let gas_used_ratio: Vec<f64> = vec![0.9; block_count_usize];
         let oldest_block: U256 = U256::from(newest_block) - _block_count;
 
-        let reward: Option<Vec<Vec<U256>>> = match _reward_percentiles {
-            Some(reward_percentiles) => {
-                let num_percentiles = reward_percentiles.len();
-                let reward_vec =
-                    vec![vec![U256::from(DEFAULT_REWARD); num_percentiles]; block_count_usize];
-                Some(reward_vec)
-            }
-            None => None,
-        };
+        let reward: Option<Vec<Vec<U256>>> = _reward_percentiles.map(|reward_percentiles| {
```

Comment:
shouldn't this be a `map_or_else`?

Reply:
Is it true that before the refactor, in case it does not match, it leaves it in None. Would this be the idea of the map_or_else in the case of the else?

Reply:
yeah, the point would be to assign `None` to `reward`,
if `_reward_percentiles` is `None`, does the current version of the code do that with `map`?

I don't even understand the refactor itself, why would we replace a match with a map here?

Reply:
All the changes applied to the PR were at Glippy's suggestion, I didn't really check if there is a unit test case where alternatives between Some and None are being tested. For now, I'm going to rollback this to avoid further blocking the PR.

Reply:
I understand that I would go through the second case of the example, I did the rollback. But let me know if you want me to change it back, sorry for the late reply.
![image](https://user-images.githubusercontent.com/58611754/222925591-60a7b360-638e-4769-94e6-cd72a0340fc8.png)

but I think the refactor makes sense, the map_or will put None taking into account the value of reward_percentiles, the rest seems to be the same.
![image](https://user-images.githubusercontent.com/58611754/222925450-0d051f33-391a-462d-b0bb-1fd430b10fa1.png)